pkg/cloudprovider/openstack/openstack.go

```diff
@@ -630,31 +631,49 @@ func (lb *LoadBalancer) UpdateTCPLoadBalancer(name, region string, hosts []strin
 	return nil
 }
 
-func (lb *LoadBalancer) DeleteTCPLoadBalancer(name, region string) error {
-	glog.V(4).Infof("DeleteTCPLoadBalancer(%v, %v)", name, region)
-
-	vip, err := getVipByName(lb.network, name)
-	if err != nil {
-		return err
-	}
-
-	pool, err := pools.Get(lb.network, vip.PoolID).Extract()
-	if err != nil {
-		return err
+func (lb *LoadBalancer) EnsureTCPLoadBalancerDeleted(name, region string) error {
+	glog.V(4).Infof("EnsureTCPLoadBalancerDeleted(%v, %v)", name, region)
+
+	// TODO: Because the we look up the pool using the VIP object, if the VIP is
+	// already gone we can't attempt to delete the pool. We should instead look up
+	// the pool by name and attempt to delete it even if the VIP has already been
+	// deleted.
+	vip, vipErr := getVipByName(lb.network, name)
+	if vipErr == ErrNotFound {
+		return nil
+	} else if vipErr != nil {
+		return vipErr
+	}
+	vipExists := (vipErr == nil)
+
+	// It's ok if the pool doesn't exist, as we may still need to delete the vip
+	// (although I don't believe the system should ever be in that state).
+	pool, poolErr := pools.Get(lb.network, vip.PoolID).Extract()
+	if poolErr != nil {
+		detailedErr, ok := poolErr.(*gophercloud.UnexpectedResponseCodeError)
+		if !ok || detailedErr.Actual != http.StatusNotFound {
+			return poolErr
+		}
 	}
+	poolExists := (poolErr == nil)
 
 	// Have to delete VIP before pool can be deleted
-	err = vips.Delete(lb.network, vip.ID).ExtractErr()
-	if err != nil {
-		return err
+	if vipExists {
```

Comment:
See above.  Unless I'm confused, this change doesn't do anything, and just complicates the code.  But I have flu, so feel free to correct me if my head is confused.